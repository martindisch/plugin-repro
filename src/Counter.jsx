import { useState } from "react";
import styled from 'styled-components'

const Heading = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
`;

const Button = styled.button`
  padding: 4em;
  background: papayawhip;
`;

export const Counter = () => {
  const [count, setCount] = useState(0);
  return (
    <>
      <Heading>{count}</Heading>
      <Button onClick={() => setCount(count + 1)}>Increment</Button>
    </>
  );
};
