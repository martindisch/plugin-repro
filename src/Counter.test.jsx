import renderer from "react-test-renderer";
import {Counter} from "./Counter";

it("renders correctly", () => {
  const tree = renderer.create(<Counter />).toJSON();
  expect(tree).toMatchSnapshot();
});
