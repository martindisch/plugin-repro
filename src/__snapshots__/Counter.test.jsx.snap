// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <h1>
    0
  </h1>,
  <button
    onClick={[Function]}
  >
    Increment
  </button>,
]
`;
